### DISCUSSION TITLE
How does the bootloader notice there is a new image in the secondary slot

### DISCUSSION THREAD

**Discussion by xianghui-renesas** [mcu-tools/mcuboot#1480] [Q&A] [answered]
Hi, when the bootloader passes the execution to the primary application, the primary application is booted. At this time, if a new image is downloaded to the secondary slot, how does the bootloader see this new image and start to validate it? Can some one help to explain this details? thanks!

Hi, the above question is resolved. There are multiple ways this can be done. It involves a reset to the system. thanks!

**Comment by d3zd3z** [comment, marked as answer]
For most configurations, there is a magic number that is written at the end of the slot that marks the slot for upgrade. At this point, at least MCUboot will then look at the slot to see if an upgrade can be done (it will try to validate the image in the second slot). Until the magic value is written, it won't even look at it. Look at `boot_set_pending` and similar functions in the public API.

**Reply by xianghui-renesas** [reply]
Thanks David for this additional information. This is very helpful!
